test/setup/index.ts

```diff
@@ -92,10 +92,13 @@ export async function deployMonitor(): Promise<string> {
 
     console.log(`platform chosen is ${testPlatform}, createCluster===${createCluster}`);
 
-    await kubectl.downloadKubectl();
+    // TODO: hack, rewrite this
+    const kubernetesVersion = testPlatform === 'openshift3' ? 'v1.11.10' : undefined;
+    await kubectl.downloadKubectl(kubernetesVersion);
```

Comment:
if we let each platform decide (or not decide) which kubectl version it wants, maybe indicated by "latest" instead of undefined, we would be able to avoid some of the code written to handle the undefined parts, wdyt?
we'd still need to handle "latest" of course but I think it could be clearer

Reply:
I find that I'll have to put another variable in the IPlatform interface to achieve this, do we really want to grow that? Or write some function `getKubernetesVersionForPlatform(testPlatform: string)` in the index.ts of the platforms module which will do a switch-case/if for all platforms?

Reply:
your call, I just really don't want ```test/setup/index.ts``` to have more and more references to each specific platform. one of the reasons we reverted the original Openshift4 PR was just that.
I really want a clean flow of events with clean abstraction as much as possible.
happy to 🍐 on that if you want